macOSUpgrade.sh

```diff
@@ -153,10 +153,38 @@ osinstallersetupdDaemonSettingsFilePath="/Library/LaunchDaemons/com.jamfps.clean
 ##Launch agent settings for filevault authenticated reboots
 osinstallersetupdAgentSettingsFilePath="/Library/LaunchAgents/com.apple.install.osinstallersetupd.plist"
 
+##Amount of time (in seconds) to allow a user to connect to AC power before moving on
+##If null or 0, then the user will not have the opportunity to connect to AC power
+acPowerWaitTimer="0"
+
+##Icon to display during the AC Power warning
+warnIcon="/System/Library/CoreServices/CoreTypes.bundle/Contents/Resources/AlertCautionIcon.icns"
+
+##Icon to display when errors are found
+errorIcon="/System/Library/CoreServices/CoreTypes.bundle/Contents/Resources/AlertStopIcon.icns"
+
 # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # #
 # FUNCTIONS
 # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # #
 
+wait_for_ac_power() {
+    # Loop for "acPowerWaitTimer" seconds until either AC Power is detected or the timer is up
+    sysRequirementErrors=()
+    /bin/echo "Waiting for AC power..."
+    while [[ "$acPowerWaitTimer" -gt "0" ]]; do
+        if /usr/bin/pmset -g ps | /usr/bin/grep "AC Power" > /dev/null ; then
+            /bin/echo "Power Check: OK - AC Power Detected"
+            /bin/ps -p "$jamfHelperPowerPID" > /dev/null && /bin/kill "$jamfHelperPowerPID"; wait "$jamfHelperPowerPID" 2>/dev/null
```

Comment:
You don't need to use ';' to put your script in one line, as we want to focus on readability.

Reply:
Replaced all kill process lines with a function, as it is used several times in the script.
View [3c0067c](https://github.com/kc9wwh/macOSUpgrade/pull/115/commits/3c0067c1bbf88e2ab4059688ad4c12a183d5dc85)